Include/Rover/Range.hpp

```diff
@@ -132,9 +136,10 @@ namespace Rover {
     auto begin = evaluator.evaluate(m_begin);
     auto end = evaluator.evaluate(m_end);
     while(true) {
-      auto random_fraction = m_distribution(m_engine);
-      auto distance = static_cast<Type>((end - begin) * random_fraction);
-      auto value = begin + distance;
+      auto value = calculate_random(begin, end);
+      if(value == begin || calculate_random(begin, end) == begin) {
```

Comment:
This looks interesting, can you explain how this works?

Reply:
It is a bit more complicated than that but I think I have solved it. I will make another commit and explain.